core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -552,68 +561,32 @@ private void displayStreamingNotAllowedNotification(Intent originalIntent) {
      * Handles media button events
      * return: keycode was handled
      */
-    private boolean handleKeycode(int keycode, boolean notificationButton) {
+    private boolean handleKeycode(int keycode, boolean mediaButton) {
         Log.d(TAG, "Handling keycode: " + keycode);
         final PlaybackServiceMediaPlayer.PSMPInfo info = mediaPlayer.getPSMPInfo();
         final PlayerStatus status = info.playerStatus;
         switch (keycode) {
             case KeyEvent.KEYCODE_HEADSETHOOK:
             case KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE:
-                if (status == PlayerStatus.PLAYING) {
-                    mediaPlayer.pause(!UserPreferences.isPersistNotify(), false);
-                } else if (status == PlayerStatus.PAUSED || status == PlayerStatus.PREPARED) {
-                    mediaPlayer.resume();
-                } else if (status == PlayerStatus.PREPARING) {
-                    mediaPlayer.setStartWhenPrepared(!mediaPlayer.isStartWhenPrepared());
-                } else if (status == PlayerStatus.INITIALIZED) {
-                    mediaPlayer.setStartWhenPrepared(true);
-                    mediaPlayer.prepare();
-                } else if (mediaPlayer.getPlayable() == null) {
-                    startPlayingFromPreferences();
-                }
+                handleKeyAction(KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE, mediaButton, MediaAction.TOGGLE_PLAYBACK);
```

Comment:
@ByteHamster when you originally tested this KEYCODE_HEADSETHOOK was not being translated as I intended. It should be fixed via c80c550214ac974f50976b961005a589460511c9